src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.Unix.cs

```diff
@@ -63,6 +63,8 @@ public static partial class PlatformDetection
         public static bool IsDrawingSupported { get; } = GetGdiplusIsAvailable();
         public static bool IsSoundPlaySupported { get; } = false;
 
+        public static bool IsDocker => RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && File.Exists("/.dockerenv");
```

Comment:
What about Windows containers? 

It seems that the `IsDocker` will return 'false' even on Windows container images such as Nano (which is always a container image) and Windows Server Core (which could be a container image or a real image)

Reply:
I was wondering about that and decided to use IsDocker instead of IsContainer. The failing tests were Linux specific.

Reply:
"IsDocker" doesn't mean "Linux".  Docker is used on Windows (i.e. Docker Desktop for Windows).  So, we should probably rename these to 'IsLinuxContainer' (which you've implemented) and 'IsWindowsContainer' (which isn't implemented yet as detection logic. not sure how you detect that anyways).

Reply:
Is there good way how to detect Docker inside Windows? Since we already have platform detection I see no good reason to split this to be platform specific.

Reply:
>Is there good way how to detect Docker inside Windows?

I don't know.   See this: https://stackoverflow.com/questions/43002803/detect-if-process-executes-inside-a-windows-container.  The best answer there seems to check for a registry key:

>There's a "ContainerType" registry value under HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control.

But if we can't figure this out, then you should rename it to 'IsLinuxContainer' instead of 'IsContainer'. Either way, it should not be named 'IsDocker'.

Reply:
I did more testing with Docker Desktop on Windows. 
One needs to choose between Linux or Windows containers. When using former, it is not possible to run Nano and derivative containers. For Linux containers current logic works even if hosted on Windows. When using Windows  containers, ContainerType is set. However, we already have detection for that: IsInAppContainer (IsInAppContainer => false) on Unix.

Maybe best option would be using IsInContainer returning IsInAppContainer for Windows and proposed change for Linux. This could be further generalized when other UNIX like OSes support docker or containerization technology.

Reply:
>Maybe best option would be using IsInContainer returning IsInAppContainer for Windows and proposed change for Linux.

"IsInAppContainer" has nothing to do with "containers" in terms of how we think of Windows Containers or Docker Containers.  "AppContainer" is a concept for UWP scenarios mostly and deals with Windows low-IL security process boundary.

So, conceptually there is no relationship between IsInAppContainer" and "IsInContainer".  It's unfortunate that the term "container" is being used in two totally different contexts.